backend/api/signals.py

```diff
@@ -28,3 +28,41 @@ def handle_stock_post_save(instance: models.Stock, created: bool, **kwargs):
                     stock=instance,
                 ),
             )
+
+        # Add Twitter
+        df = pd.DataFrame()
+        data = services.get_tweets(instance.ticker)
+        df = pd.concat([df, data])
+        for args in df.itertuples(index=False):
+            models.Tweet.objects.update_or_create(
+                api_id=0,  # TODO make api_id schema unique in models.py
+                defaults=dict(
```

Comment:
Should be `api_id=args.<name of api id col in df>`

The way `update_or_create` works is that it will filter based on the `**kwargs` minus `defaults` so in this case `api_id`. If it finds a match it will update the record else it will create a new object with whatever is in `defaults` for both updating and creating. With this implementation it will only ever have 1 tweet generated.